### DISCUSSION TITLE
page up/down moves through cmd history, instead of scrolling

### DISCUSSION THREAD

**Discussion by mustafa0x** [ghostty-org/ghostty#3875] [Q&A] [answered]
and if i press shift+pg+up/down, `2~` is printed

**Comment by 00-kat** [comment, marked as answer]
Those keys are sent to the shell.  You could bind them yourselves:
```ini
keybind = page_up=scroll_page_up
keybind = page_down=scroll_page_down
keybind = shift+page_up=adjust_selection:page_up
keybind = shift+page_down=adjust_selection:page_down
```
The latter two don't seem to work for me, but they *are* listed in the source code, so I'm not sure about that (I see `warning(gtk): unimplemented action=apprt.action.Action.Key.key_sequence` when it's triggered).

**Reply by mustafa0x** [reply]
Thanks. Very strange. It was working fine before. I didn't change anything.

Those keybindings help, but not perfectly.

Eg, if I open fzf via ctrl+r, page up doesn't navigate history. adding shift doesn't help.

**Reply by 00-kat** [reply]
> page up doesn't navigate history

Yes, because those rebind them.  This is something a shell would do, so consider checking your shell's documentation on how to rebind things instead of rebinding it in Ghostty.

> adding shift doesn't help

What is adding shift meant to do?  I assumed you wanted it to select an entire page, which is why I added the last two lines in that snippet.

**Reply by J-Siu** [reply]
Thank you! I didn't know there are `scroll_page_*` as they are not listed in `+list-actions` .

I end up with following to match iTerm2 and Linux behavior:
```
keybind = shift+page_down=scroll_page_down
keybind = shift+page_up=scroll_page_up
```

**Comment by knirch** [comment]
Sorry for butting in, but; your intention is to swap behavior?

It's my understanding "traditionally"  Page Up/Page Down scrolls command history and with shift the terminal emulator scrolls.

You want to reverse that? So that Ghostty scrolls on page up/ page down and readline/bash reacts to shift-page up,shift-page down?

If that's true; then (unless there's something weird in ghostty, still coming to grips with it stomping in with bashpreexec etc :D) you probably want to adjust your inputrc?   Couldn't find a list of fzf default binds, so no idea if you need to tweak that too.

(I realize I just assumed bash/readline, no idea what other shells use)

**Reply by mustafa0x** [reply]
> It's my understanding "traditionally" Page Up/Page Down scrolls command history and with shift the terminal emulator scrolls.

The Mac terminal is the opposite. pg up/down scrolls position, and shift + pg up/down scrolls cmd history. Ghostty did the same when first installed. Then after a few days of usage, its behavior changed as described above. I don't remember changing anything. I don't have an inputrc file.

**Comment by etiennebarrie** [comment]
I was also bitten by that, coming from macOS Terminal.app. I realize that inside screen/tmux, most terminals, this is how it works, so I'm willing to change my habits. Particularly since I quickly found out that one can use command+page up/down to scroll the view.
